crates/ott-common/src/discovery/dns.rs

```diff
@@ -32,7 +32,10 @@ impl DnsServiceDiscoverer {
 impl ServiceDiscoverer for DnsServiceDiscoverer {
     async fn discover(&mut self) -> anyhow::Result<Vec<ConnectionConfig>> {
         let resolver =
-            TokioAsyncResolver::tokio_from_system_conf().expect("failed to create resolver");
+            match self.config.dns_server {
+                None => TokioAsyncResolver::tokio_from_system_conf().expect("failed to create resolver"),
+                Some(server) => TokioAsyncResolver::tokio(ResolverConfig::new().set_domain(server), ResolverOpts::default()).expect("failed to create resolver"),
```

Comment:
I don't think `set_domain` does what we want here. [`add_name_server`](https://docs.rs/trust-dns-resolver/latest/trust_dns_resolver/config/struct.ResolverConfig.html#method.add_name_server) is probably what you want.

You also might want to change `dns_server` from a `String` to a `SocketAddr` or `IpAddr`.